ipaserver/plugins/service.py

```diff
@@ -568,6 +568,27 @@ def validate_ipakrbauthzdata(self, entry):
             raise errors.ValidationError(name='ipakrbauthzdata',
                 error=_('NONE value cannot be combined with other PAC types'))
 
+    def validate_auth_indicator(self, entry):
+        new_value = entry.get('krbprincipalauthind', None)
+
+        if not new_value:
+            return
+
+        # The following services are considered internal IPA services
+        # and shouldn't be allowed to have auth indicators.
+        # https://pagure.io/freeipa/issue/8206
+        server_services = ('host', 'ldap', 'HTTP', 'cifs')
+        client_services = ('host', 'cifs')
+        services = server_services if api.env.in_server else client_services
+        name = unicode(self.get_primary_key_from_dn(entry.dn))
+        for service in services:
+            if service + '/' in name:
```

Comment:
Explanation: get_primary_key_from_dn() returns a str (text). str.startswith() also accepts a tuple of strings.

Also ``ipa.env.in_server`` is wrong. Since the code is executed in ipaserver code, the flag is always true. You have to convert entry to a ``ipapython.kerberos.Principal`` object, then check if the hostname is an IPA server with ``host_is_master()`` or ``server_find()``.